### PR TITLE
meson: handle nested disabler

### LINKED ISSUE #5659: Disabler in nested arrays not working fine
See #5658

### PR DESCRIPTION
**Pull request by elmarco**
Fixes #5659
Signed-off-by: Marc-André Lureau <marcandre.lureau@redhat.com>

### CODE REVIEW

File: mesonbuild/interpreterbase.py [hunk 1]
```diff
@@ -340,17 +340,21 @@ def found_method(self, args, kwargs):
 def is_disabler(i):
     return isinstance(i, Disabler)
 
+def is_arg_disabled(arg):
+    if is_disabler(arg):
+        return True
+    if isinstance(arg, list):
+        for i in arg:
+            if is_arg_disabled(i):
+                return True
```
Comment (dcbaker) [inline mesonbuild/interpreterbase.py]: I think ti would be better to return an explicit `False` at the end fo the function, and add a return annotation `bool`

Comment (elmarco) [reply mesonbuild/interpreterbase.py]: good idea, done

**Comment by nirbheek** [comment]
I'm going to pull this into the 0.51.2 stable release, seems like a useful enough fix to get out quickly.

**Comment by nirbheek** [comment]
CI is failing spuriously over and over. Will retry after a few hours.

**Comment by dcbaker** [comment]
Those build failures look like for some reason powershell is parsing the pip output and noticing the warning and causing an error. since this passed before adding `return False`, I'm going to call it good and merge it.

**Comment by QuLogic** [comment]
> powershell is parsing the pip output and noticing the warning and causing an error

Yes, Powershell (perhaps only on Azure) does that... Because exit codes are not trustworthy on Windows, or something.

**Comment by dcbaker** [comment]
There's a pip argument to not look for newer versions of pip IIRC, we should probably be passing that?

**Comment by nirbheek** [comment]
> There's a pip argument to not look for newer versions of pip IIRC, we should probably be passing that?

https://github.com/mesonbuild/meson/pull/5705

**Pull request merged**